Src/FluentAssertions/AsyncAssertionsExtensions.cs

```diff
@@ -22,6 +23,11 @@ public static class AsyncAssertionsExtensions
         this Task<AndWhichConstraint<GenericAsyncFunctionAssertions<T>, T>> task,
         T expected, string because = "", params object[] becauseArgs)
     {
+        if (AssertionScope.Current.HasFailures())
+        {
+            return null;
+        }
```

Comment:
Other things than `CompleteWithinAsync` could have added a failure to the `AssertionScope`.

```cs
using var _ = new AssertionScope();
true.Should().BeFalse();
await func.Should().CompleteWithinAsync(100.Milliseconds()).WithResult(2);
```

Reply:
Yes, I know.
But what is the consequence? We skip one check while another already caused a failed test.

This is just a prove of concept. I'm thinking about a solution to prevent from test not completing.

Reply:
The whole idea of `AssertionScope` is to continue asserting where possible.
So if `true.Should().BeFalse();` fails, but `CompleteWithinAsync` succeeds, `WithResult` should be executed.

But I get the case you want to solve, just wanted to drop a note.